src/Microsoft.VisualStudio.Jdt.Tests/JsonTransformationTest.cs

```diff
@@ -275,6 +277,41 @@ public void ThrowAndLogException()
             }
         }
 
+        /// <summary>
+        /// Tests that a transformation succeeds even if the source and transform files are read-only.
+        /// </summary>
+        [Fact]
+        public void ReadOnly()
+        {
+            const string TransformSourceString = @"{
+                                                        '@jdt.rename' : {
+                                                            'A' : 'Astar'
+                                                        }
+                                                    }";
+
+            // create temporary files to use for the source and transform
+            using (ReadOnlyTempFile tempSourceFilePath = new ReadOnlyTempFile(SimpleSourceString))
+            using (ReadOnlyTempFile tempTransformFilePath = new ReadOnlyTempFile(TransformSourceString))
+            {
+                // apply transform
+                JsonTransformation transformation = new JsonTransformation(tempSourceFilePath.FilePath, this.logger);
+                using (Stream result = transformation.Apply(tempTransformFilePath.FilePath))
+                using (StreamReader streamReader = new StreamReader(result))
+                using (JsonTextReader jsonReader = new JsonTextReader(streamReader))
+                {
+                    var transformed = JObject.Load(jsonReader);
+                    var expected = JObject.Parse(@"{
```

Comment:
Since you renamed A, the expected result should be:
`
{
    'Astar': 1
}
`
In any case, this test doesn't need to (and probably shouldn't) verify the correctness of the transformation. You can just verify that the result stream is not null and no exceptions were recorded, like in `TryTransformTest`.